packages/spatial/src/renderer/components/PostProcessingComponent.tsx

```diff
@@ -59,16 +104,532 @@ export const PostProcessingComponent = defineComponent({
-      )
-    }, [rendererEntity, postprocessingComponent.enabled, postprocessingComponent.effects])
+    if (!rendererEntity) return null
 
-    return null
+    return <RendererReactor entity={entity} rendererEntity={rendererEntity} />
   }
 })
+
+const RendererReactor = (props: { entity: Entity; rendererEntity: Entity }) => {
+  const { entity, rendererEntity } = props
+
+  const postprocessingComponent = useComponent(entity, PostProcessingComponent)
+  const camera = useComponent(rendererEntity, CameraComponent)
+  const renderer = useComponent(rendererEntity, RendererComponent)
+  const composer = new EffectComposer(renderer.value.renderer)
+  renderer.value.effectComposer = composer
+  const renderPass = new RenderPass()
+  renderer.value.effectComposer.addPass(renderPass)
+  renderer.value.renderPass = renderPass
+
+  let lut1DEffectTexturePath: string | undefined
+  if (
+    postprocessingComponent.effects[Effects.LUT1DEffect].lutPath &&
+    postprocessingComponent.effects[Effects.LUT1DEffect].isActive.value
+  ) {
+    lut1DEffectTexturePath = postprocessingComponent.effects[Effects.LUT1DEffect].lutPath.value
+  }
+  let lut3DEffectTexturePath: string | undefined
+  if (
+    postprocessingComponent.effects[Effects.LUT3DEffect].lutPath &&
+    postprocessingComponent.effects[Effects.LUT3DEffect].isActive.value
+  ) {
+    lut3DEffectTexturePath = postprocessingComponent.effects[Effects.LUT3DEffect].lutPath.value
+  }
+  let textureEffectTexturePath: string | undefined
+  if (
+    postprocessingComponent.effects[Effects.TextureEffect].texturePath &&
+    postprocessingComponent.effects[Effects.TextureEffect].isActive.value
+  ) {
+    textureEffectTexturePath = postprocessingComponent.effects[Effects.TextureEffect].texturePath.value
+  }
+
+  const [lut1DEffectTexture, lut1DEffectTextureError] = useTexture(lut1DEffectTexturePath!, entity)
+  const [lut3DEffectTexture, lut3DEffectTextureError] = useTexture(lut3DEffectTexturePath!, entity)
+  const [textureEffectTexture, textureEffectTextureError] = useTexture(textureEffectTexturePath!, entity)
+
+  const scene = useHookstate<Scene>(() => new Scene())
+  const normalPass = useHookstate<CustomNormalPass>(() => new CustomNormalPass(scene, camera))
+  const depthDownsamplingPass = useHookstate<DepthDownsamplingPass>(
+    () =>
+      new DepthDownsamplingPass({
+        normalBuffer: normalPass.value.texture,
+        resolutionScale: 0.5
+      })
+  )
+  const velocityDepthNormalPass = useHookstate(new VelocityDepthNormalPass(scene, camera))
+  const useVelocityDepthNormalPass = useHookstate(false)
+  const useDepthDownsamplingPass = useHookstate(false)
+
+  //const composer = useHookstate<EffectComposer>(() => new EffectComposer(renderer.value.renderer))
+  const effects = useHookstate<Record<string, Effect>>({})
+
+  useEffect(() => {
+    velocityDepthNormalPass.set(new VelocityDepthNormalPass(scene, camera))
+  }, [scene])
+
+  useEffect(() => {
```

Comment:
i think a cleaner pattern for specific effects would be to define a dictionary of effects to create/destroy functions that are invoked, rather than hardcoding each effect. we can then use sub-reactors to map each effect into it's own react component, and handle create/destroy/update events in a controlled and generic way.